.travis.yml

```diff
@@ -18,10 +33,14 @@ install:
         (cd libsodium && ./configure --prefix=$HOME/.ghc/usr && make install -j$(nproc))
       }
   # hs-toxcore
-  - cabal update
-  - >
-      [ -f $HOME/.cabal/bin/happy ] || cabal install happy
   - cabal install hstox --extra-include-dirs=$HOME/.ghc/usr/include --extra-lib-dirs=$HOME/.ghc/usr/lib
 
 script:
   - make
+
+branches:
+  only:
+  - master
+
+git:
+  depth: 1
```

Comment:
not sure if this still is the case but I have seen restart of builds fail when someone has already pushed other commits, so the commit to check out for the build is not included in "depth 1".